pkg/kapp/clusterapply/waiting_changes.go

```diff
@@ -62,13 +65,19 @@ func (c *WaitingChanges) WaitForAny() ([]WaitingChange, error) {
 		for _, change := range c.trackedChanges {
 			change := change // copy
 
-			go func() {
+			go func(timeout time.Duration) {
+				//To start tracking the time for the new resource
+				if change.startTime.IsZero() {
+					change.startTime = time.Now()
+				}
+
 				waitThrottle.Take()
 				defer waitThrottle.Done()
 
 				state, descMsgs, err := change.Cluster.IsDoneApplying()
+				err = checkForTimeout(change, timeout, err)
```

Comment:
lets inline logic for checkForTimeout here.

Reply:
Done